packages/deployments/contracts/contracts/core/connext/facets/RoutersFacet.sol

```diff
@@ -249,86 +237,47 @@ contract RoutersFacet is BaseConnextFacet {
    * @param _router The relevant router address
    */
   function getRouterApprovalForPortal(address _router) public view returns (bool) {
-    return s.routerPermissionInfo.approvedForPortalRouters[_router];
+    return s.routerConfigs[_router].portalApproved;
   }
 
   // ============ Admin methods ==============
 
   /**
-   * @notice Used to set router initial properties
-   * @param router Router address to setup
-   * @param owner Initial Owner of router
-   * @param recipient Initial Recipient of router
+   * @notice Used to whitelist a given router
+   * @param _router Router address to setup
    */
-  function setupRouter(
-    address router,
-    address owner,
-    address recipient
-  ) external onlyOwnerOrRouter {
+  function addRouter(address _router) external onlyOwnerOrRouter {
     // Sanity check: not empty
-    if (router == address(0)) revert RoutersFacet__setupRouter_routerEmpty();
+    if (_router == address(0)) revert RoutersFacet__addRouter_routerEmpty();
 
     // Sanity check: needs approval
-    if (s.routerPermissionInfo.approvedRouters[router]) revert RoutersFacet__setupRouter_alreadyAdded();
+    if (s.routerConfigs[_router].approved) revert RoutersFacet__addRouter_alreadyAdded();
```

Comment:
Perhaps check the router is initialized?
e.g. `s.routerConfigs[_router].owner != address(0)`

Reply:
i thought about it, but dont see a reason that admins cant preapprove a router. if the EOA was added incorrectly, that's a concern regardless (i.e. typos).

Reply:
agreed. approval should we be a separate switch to configuration, which should be done at the EOA owner's discretion. i.e. those steps should be basically async